lib/hardware-mapping.c

```diff
@@ -146,6 +146,93 @@ struct HardwareMapping matrix_hardware_mappings[] = {
     /* No more chains - there are not enough GPIO */
   },
 
+  /*
+   * The mapping for JI's hat:
+   * https://github.com/jayeye/ji/src/kicad/rpi-rgb-led-matrix-ji
+   */
+  {
+    .name          = "ji-hat",
+
+    .output_enable = GPIO_BIT(18),
+    .clock         = GPIO_BIT(27),
+    .strobe        = GPIO_BIT(21),
+
+    /* Address lines */
+    .a             = GPIO_BIT(16),
+    .b             = GPIO_BIT(19),
+    .c             = GPIO_BIT(20),
+    .d             = GPIO_BIT(26),
+    .e             = GPIO_BIT(13),
+
+    /* Parallel chain 0, RGB for both sub-panels */
+    .p0_r1         = GPIO_BIT(4),
+    .p0_g1         = GPIO_BIT(15),
+    .p0_b1         = GPIO_BIT(14),
+    .p0_r2         = GPIO_BIT(3),
+    .p0_g2         = GPIO_BIT(17),
+    .p0_b2         = GPIO_BIT(2),
+
+    /* Chain 1 */
+    .p1_r1         = GPIO_BIT(10),
+    .p1_g1         = GPIO_BIT(25),
+    .p1_b1         = GPIO_BIT(24),
+    .p1_r2         = GPIO_BIT(22),
+    .p1_g2         = GPIO_BIT(9),
+    .p1_b2         = GPIO_BIT(23),
+
+    /* Chain 2 */
+    .p2_r1         = GPIO_BIT(5),
+    .p2_g1         = GPIO_BIT(12),
+    .p2_b1         = GPIO_BIT(7),
+    .p2_r2         = GPIO_BIT(8),
+    .p2_g2         = GPIO_BIT(6),
+    .p2_b2         = GPIO_BIT(11),
+  },
+
+  /*
+   * The mapping for JI's hat(s):
+   * https://github.com/jayeye/ji/src/kicad/rpi-rgb-led-matrix-ji
+   * The first batch of the zero adaptor had swapped G1/G2 lines.
```

Comment:
Given that this was possibly just a test-run with only you or a few other having seen the hardware, do you really want to include the buggy version of the adapter in the main rpi-rgb-led-matrix distribution ?